src/nvim/screen.c

```diff
@@ -2403,10 +2403,10 @@ win_line (
   filler_todo = filler_lines;
 
   // Cursor line highlighting for 'cursorline' in the current window.
-  if (wp->w_p_cul && lnum == wp->w_cursor.lnum) {
+  if (lnum == wp->w_cursor.lnum && win_cursorline_standout(wp)) {
```

Comment:
Perhaps we can just update `wp->w_last_cursorline` unconditionally, I don't think anything is positively gained by keeping around the stale value even with no "standout" features active.

Reply:
Sounds good (changed).
I was under the impression that `w_last_cursorline` would be used with `w_p_cul` only, but that's changed since 23c71d5 / #9492 at least.